gramex/data.py

```diff
@@ -1460,9 +1477,13 @@ def _mongodb_json(obj):
 def _filter_mongodb(url, controls, args, database=None, collection=None, query=None, **kwargs):
     '''TODO: Document function and usage'''
     table = _mongodb_collection(url, database, collection, **kwargs)
-    if query is None:
-        meta_cols = pd.DataFrame(list(table.find().limit(100)))
-        query = _mongodb_query(args, meta_cols)
+
+    results = table.find().limit(100)
+    meta_cols = pd.DataFrame(list(copy.deepcopy(results)))
+    if query is not None:
+        query = dict(query)
+    else:
+        query = _mongodb_query(args, meta_cols, results=results)
```

Comment:
Again, another possibility. Don't use Pandas to read the first few rows. Instead, use pymongo & detect the type directly. This may be better.